webapp/components/create_team/components/display_name.jsx

```diff
@@ -118,14 +117,6 @@ export default class TeamSignupDisplayNamePage extends React.Component {
                             defaultMessage='Next'
                         /><i className='fa fa-chevron-right'/>
                     </button>
-                    <div className='margin--extra'>
-                        <Link to='/select_team'>
-                            <FormattedMessage
-                                id='create_team.display_name.back'
```

Comment:
I think we also need to get rid of i18n strings in "webapp/i18n/*.json" as "create_team.display_name.back" is the only user of this string.